src/unix/linux-core.c

```diff
@@ -1015,14 +1025,262 @@ uint64_t uv_get_total_memory(void) {
+    if (getline(&buf, &buf_size, fp) < 0) {
+      if (feof(fp) != 0)
+        break;
+      goto file_malformed;
+    }
+    if ('\n' == buf[strlen(buf) - 1])
+      buf[strlen(buf) - 1] = '\0';
+
+    field_ptr = buf;
+    strsep(&field_ptr, " "); /* mount ID */
+    strsep(&field_ptr, " "); /* parent ID */
+    strsep(&field_ptr, " "); /* st_dev major:minor */
+    curr_root = strsep(&field_ptr, " ");
+    curr_mount_point = strsep(&field_ptr, " ");
+    strsep(&field_ptr, " "); /* mount options */
+    /* A hyphen marks the end of variable-length optional fields. */
+    while ('-' != field_ptr[0])
+      strsep(&field_ptr, " ");
+    strsep(&field_ptr, " "); /* separator (hyphen) */
+    curr_fs_type = strsep(&field_ptr, " ");
+    strsep(&field_ptr, " "); /* mount source */
+    curr_super_options = strsep(&field_ptr, " ");
+    /*
+     * If the fs type (9) is "cgroup" and super options (11) contains the name
+     * of the subsystem, then we've found the correct mount location, so save
+     * the values of root (4) and mount point (5) and break.
+     * Otherwise, if the fs type is "cgroup2", we've potentially found the
+     * correct mount location, so save the above values, but don't break,
+     * because we don't know yet whether the input subsystem is controlled by
+     * cgroups v1 or v2.
+     */
+    if (strcmp(curr_fs_type, "cgroup") == 0) {
+      /* cgroups v1 */
+      if (uv__find_in_delimited_string(curr_super_options, subsystem, ",") != 0) {
+        uv__strscpy(root, curr_root, sizeof(root));
+        uv__strscpy(mount_point, curr_mount_point, sizeof(mount_point));
+        info->cgroups_version = CGROUPS_VERSION_1;
+        break;
+      }
+    } else if (strcmp(curr_fs_type, "cgroup2") == 0) {
+      /* cgroups v2 */
+      uv__strscpy(root, curr_root, sizeof(root));
+      uv__strscpy(mount_point, curr_mount_point, sizeof(mount_point));
+      info->cgroups_version = CGROUPS_VERSION_2;
+      /*
+        * Don't break, as we're not certain that this subsystem is controlled
+        * by cgroups v2.
+        */
+    }
+  }
+
+  fclose(fp);
+  uv__free(buf);
+  buf_size = 0;
+
+  /*
+   * If cgroups version wasn't determined, assume this subsystem isn't enabled
+   * in cgroups, so don't bother reading /proc/self/cgroup.
+   */
+  if (CGROUPS_VERSION_UNKNOWN == info->cgroups_version)
+    return 0;
+  
+  /* Read /proc/self/cgroup to get hierarchy path. */
+
+  hierarchy_path[0] = '\0';
+
+  fp = uv__open_file(PROC_SELF_CGROUP);
+  if (fp == NULL)
+    return UV__ERR(errno);
+  
+  /* + 3 for two colons, and terminal character. */
+  subsystem_search_string = uv__malloc(strlen(subsystem) + 3);
+  snprintf(subsystem_search_string, strlen(subsystem) + 3, ":%s:", subsystem);
+
+  while (feof(fp) == 0) {
+    if (getline(&buf, &buf_size, fp) < 0) {
+      if (feof(fp) != 0)
+        break;
+      goto file_malformed;
+    }
+    if ('\n' == buf[strlen(buf) - 1])
+      buf[strlen(buf) - 1] = '\0';
```

Comment:
If you cache the result of `getline()`, you don't need to call `strlen()` twice. (Also applies to line 1155-1156.)